pxtblocks/fields/field_melodySandbox.ts

```diff
@@ -299,6 +328,47 @@ namespace pxtblockly {
                 if (this.tempoInput) {
                     this.tempoInput.value = this.tempo + "";
                 }
+                // update tempo on block if not a variable
+                const s = this.sourceBlock_;
+                if (s.parentBlock_) {
```

Comment:
Should this just be a call to syncTempoField?

Reply:
One is syncing the field on the block with the field in the dropdown editor and the other is syncing from the editor to the field on the block. There is a lot of overlap in code though so I'll edit the sync function so that it can do both.